src/primitives/transaction.h

```diff
@@ -357,6 +426,9 @@ class CTransaction
     const std::vector<CTxOut> vout;
     const uint32_t nLockTime;
     const uint32_t nExpiryHeight;
+    const CAmount valueBalance;
+    const std::vector<SpendDescription> vShieldedSpend;
+    const std::vector<OutputDescription> vShieldedOutput;
```

Comment:
Note to reviewers that this is missing the binding signature, but that can be added in another PR. (Or, add it to this PR. It's just going to be a `base_blob<512>`.)

Reply:
Will add (using `boost::array` in this case, to match `joinSplitSig`).